### ISSUE TITLE
Adding DEBUG mode for tracking complete user agent

### ISSUE THREAD

**Issue by Zodiac1978** [pluginkollektiv/statify#113]
I would like to add a Debug mode (just for us and Power Users who know what they do).

This Debug mode needs to be enabled per filter or constant and is not for the casual user. This debug mode would save the complete user agent string in a new column for testing purposes (For example: do we track bots?)

Is this too risky? If yes, this could be our use case for adding more info about the visit per custom plugin mentioned in another issue.

**Comment by Zodiac1978** [comment]
Plugin solution here: https://gist.github.com/Zodiac1978/8fbf5d67063cb213f530812d0f5b247d
But this does not work with JS tracking at the moment, see #114

**Comment by Zodiac1978** [comment]
I did some research and this would not be GDPR compliant. Tracking the user agent string seems to be a personally identifiable information. 🙄

Too risky for an official option, I think.

**Comment by stkjj** [comment]
Not necessarily. You could argue with ErwG 47 "Legitimate interest". Esp. if not a default option and with the high hurdles of filter/constant.

**Comment by stklcode** [comment]
Besides that there a a lot of countries in the world that do not know anything about GDPR and similar regulartions and/or the WP instance might be within a private network anyway. I can't see any issues against adding this as an Opt-In feature here. Probably write a note that this function might violate legal regulations and add a sentencve to the "is this plugin GDPR compliant" section of the FAQ, but finally it's down to the site admin.

**Comment by Zodiac1978** [comment]
> You could argue with ErwG 47 "Legitimate interest".

Maybe, but this had to be added to the privacy policy if activated. At the moment the user do not  need to add anything to the privacy policy.

> Probably write a note that this function might violate legal regulations and add a sentencve to the "is this plugin GDPR compliant" section of the FAQ, but finally it's down to the site admin.

Yes, with documentation and notices this could work. 

> Esp. if not a default option and with the high hurdles of filter/constant.

Yes, that could be true and then it would work. Thanks for the discussion! 👍

**Comment by stkjj** [comment]
> Besides that there a a lot of countries in the world that do not know anything about GDPR and similar regulartions …

Week argue IMHO, as things are subject to be changed. GDPR is widely discussed worldwide and local implementations are in some cases are not far away. I'ld rather would take this issue away from GDPR compliance only to privacy by design. And though it might be an edge case the "legitimate interest" could be the one and only approach to keep up with both privacy (by design) and enriched functionality.

**Comment by stklcode** [comment]
Definitely, just wanted to point out that GDPR is not the best argument against this feature either. I'm 100% with you when it comes to "priacy by design".

However it's at most 3 LOC to add the UA string to the dataset utilizing the existing `statify__visit_saved` hook right now, so adding a more convenient opt-in hook would just enable some more less code-enthusiastic admins to analyze site usage.

We could discuss little further to break the UA strings down to major 4-5 browser classes (i.e. Firefox, Chrome, Safari, IE, Edge, other) wich reveals less specific, but yet useful information. I personally can't find any real argument against lowering the bar for such feature and add a switch it to the settings page. (not for the initial raw data approach of course)
